packages/web-forms/src/components/form-layout/FormQuestion.vue

```diff
@@ -25,15 +25,33 @@ const isRangeNode = (node: ControlNode) => node.nodeType === 'range';
 const isTriggerNode = (node: ControlNode) => node.nodeType === 'trigger';
 const isUploadNode = (node: ControlNode) => node.nodeType === 'upload';
 
-const submitPressed = inject<boolean>('submitPressed', false);
+const submitPressed = inject<Ref<boolean>>('submitPressed', ref(false));
+
+const touched = ref(false);
+const stopWatch = watch(
+	() => props.question.currentState.instanceValue,
+	() => {
+		touched.value = true;
+		stopWatch();
+	},
+	{ deep: true }
+);
+
+const questionHasError = computed(() => {
+	return (
+		(touched.value || submitPressed.value) &&
+		props.question.validationState.violation?.valid === false
+	);
+});
+provide('questionHasError', questionHasError);
```

Comment:
The evaluation is done here and shared with any other child component that needs to know if the question has errors.